server/models/Document.js

```diff
@@ -427,6 +433,26 @@ Document.addHook('afterCreate', async model => {
 
 // Instance methods
 
+Document.prototype.toMarkdown = function() {
+  const text = unescape(this.text);
+
+  if (this.version) {
+    return `# ${this.title}\n\n${text}`;
+  }
+
+  return text;
+};
+
+Document.prototype.migrateVersion = function() {
+  // migrate from document version 0 -> 1 means removing the title from the
+  // document text attribute.
+  if (!this.version) {
+    this.text = this.text.replace(/^#\s(.*)\n/, '');
+    this.version = 1;
+    return this.save({ silent: true, hooks: false });
+  }
+};
```

Comment:
The strategy is to add another case in these migration functions for the eventual prosemirror migration?

Reply:
If it's required, yep.